EIPS/eip-5409.md

```diff
@@ -0,0 +1,65 @@
+created: 2022-07-23
+requires: 165, 1155
+---
+
+## Abstract
+
+This standard is an extension of [EIP-1155](./eip-1155.md). It proposes an additional function, `ownerOf` to allow EIP-1155 to support Non-Fungible Tokens (tokens that have a unique owner). By implementing this extra function, EIP-1155 can benefit from [EIP-721](./eip-721.md) core functionality without implementing the full (and less efficient) EIP-721 spec in the same contract. In particular, this extension allows EIP-1155 tokens of supply one to have an owner and have it exposed.
+
+## Motivation
+
+Currently, EIP-1155 does not allow an external caller to detect whether a token is truly unique (can have only one owner) or fungible. This is because EIP-1155 do not expose a mechanism to detect whether a token will have its supply remain to be "1". Furthermore, it does not let an external caller retrieve the owner directly on-chain.
+
+The EIP-1155 spec does mention the use of split id to represent non-fungible tokens, but this requires a pre-established convention that is not part of the standard. Plus it would not be as fully featured as the `ownerOf` function also present in the EIP-721 standard.
+
+The ability to get the owner of a token offer great benefit, including the ability for the owner to associate data to it.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Contract Interface
+
+```solidity
+interface IERC1155OwnerOf {
+
+    /// @notice Find the owner of an NFT
+    /// @dev The zero address indicates that there is no owner: either the token does not exist or it is not an NFT (supply potentially bigger than 1)
+    /// @param tokenId The identifier for an NFT
+    /// @return The address of the owner of the NFT
+    function ownerOf(uint256 tokenId) external view returns (address);
+}
+```
+
+The `ownerOf(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `supportsInterface` method MUST return `true` when called with `0x6352211e`.
+
+## Rationale
+
+Having the ability to fetch the owner for EIP-1155 tokens that have one, allows us to use them in most of the EIP-721 ecosystem, without the need to implement the whole EIP-721 spec which comes with inefficiencies.
+
+`ownerOf` does not throw when a token does not exist (or does not have an owner). This simplifies the handling of such a case. Since it would be a security risk to assume all EIP-721 implementation would throw, it should not break compatibility with contract handling EIP-721 when dealing with this EIP-1155 extension.
+
+## Backwards Compatibility
+
+It is fully backwards compatible with EIP-1155.
+
+There is no intention to support EIP-721 here, but the use of the same function `ownerOf` should allow many applications tied to EIP-721 to be compatible with EIP-1155, but not all.
+
+## Reference Implementation
+
```

Comment:
```suggestion
```

Reply:
would this not make the bot complain ?

Reply:
Nope. Reference implementation is optional.